mlir/lib/Dialect/Transform/IR/TransformInterfaces.cpp

```diff
@@ -1021,9 +1023,10 @@ transform::TransformState::applyTransform(TransformOpInterface transform) {
       // pre-generated error messages, so we do not need the association to
       // still be there when the invalidated handle is accessed.
       SmallVector<Value> handles;
-      (void)getHandlesForPayloadOp(op, handles);
+      (void)getHandlesForPayloadOp(op, handles, /*includeOutOfScope=*/true);
       for (Value handle : handles)
-        forgetMapping(handle, /*origOpFlatResults=*/ValueRange());
+        forgetMapping(handle, /*origOpFlatResults=*/ValueRange(),
+                      /*allowOutOfScope=*/true);
       cachedNames.erase(op);
```

Comment:
Should these be unconditionally `true`? Looks like we'd want to forward `allowOutOfScope`.

A test for value handle would help.

Reply:
Yes, this should always be `true`. There is no `allowOutOfScope` to forward here. (Note this is in `applyTransform`, the short diff rendering makes it look like it's in  `forgetValueMapping`, but it's not.)

I'm going to add a test for value handles.

Reply:
Actually, the `allowOutOfScope` is not needed for `forgetValueMapping` because there is no name caching for value mappings. I removed that part again.